src/streaming-client.ts

```diff
@@ -0,0 +1,115 @@
+import * as fs from 'fs';
+import * as sockets from 'websocket';
+import * as events from 'events';
+import { Duplex, PassThrough } from 'stream';
+
+import AudioConfig from './models/streaming/AudioConfig';
+import BufferedDuplex from './models/streaming/BufferedDuplex';
+import StreamingResponse from './models/streaming/StreamingResponse';
+
+export class RevAiStreamingClient extends events.EventEmitter {
+    baseUrl: string;
+    client: sockets.WebSocketClient;
+    private accessToken: string;
+    private config: AudioConfig;
+    private requests: PassThrough;
+    private responses: PassThrough;
+    constructor(accessToken: string, config: AudioConfig, version = 'v1alpha'){
+        super();
+        this.accessToken = accessToken;
+        this.config = config;
+        this.baseUrl = `wss://api.rev.ai/speechtotext/${version}/stream?` +
+            `access_token=${accessToken}` +
+            `&content_type=${config.getContentTypeString()}`;
+        this.requests = new PassThrough();
+        this.responses = new PassThrough({ objectMode: true });
+        this.client = new sockets.client();
+    }
+
+    start(): Duplex {
+        this.setUpClient();
+        this.client.connect(this.baseUrl);
+        return new BufferedDuplex(this.requests, this.responses, { readableObjectMode: true });
+    }
+
+    end(): void {
+        this.client.abort();
+        this.requests.write("EOS");
+        this.requests.end();
+        this.responses.push(null);
+    }
+
+    private setUpClient(): void {
+        var self = this;
+        this.client.on('httpResponse', function(response, client) {
+            self.emit('httpResponse', response.statusCode);
+            self.end();
+        });
+        this.client.on('connectFailed', function(error: Error) {
+            self.emit('connectFailed', error);
+            self.end();
+        });
+        this.client.on('connect', function(connection: any) {
+            connection.on('error', function(error : Error) {
+                self.emit('error', error);
+                self.requests.end();
+                self.responses.push(null);
+            });
+            connection.on('close', function(code: number, reason: string) {
+                self.emit('close', code, reason);
+                self.end();
+            });
+            connection.on('message', function(message: any) {
+                if (message.type === 'utf8') {
+                    var response = JSON.parse(message.utf8Data) as StreamingResponse;
+                    if (response.type == "connected"){
+                        self.emit('connect', response.id);
```

Comment:
I would actually return the whole `StreamingResponse` and maybe split up `StreamingResponse` into 2 classes (or 3, with 1 as a base). My reasoning is it's unclear what to parse for using the SDK as it currently is for each response. 

If we include `type` back in every response, it's easier to document and code-wise to check which message is what